staging/src/k8s.io/client-go/util/workqueue/queue_test.go

```diff
@@ -159,3 +159,72 @@ func TestReinsert(t *testing.T) {
 		t.Errorf("Expected queue to be empty. Has %v items", a)
 	}
 }
+
+func TestFinishProcessingBeforeShutdown(t *testing.T) {
+	q := workqueue.New()
+	q.Add("foo")
+	q.Add("bar")
+
+	firstItem, _ := q.Get()
+	secondItem, _ := q.Get()
+
+	finishedShutdownTest := finishedShutdownTest{}
+	go func() {
+		q.ShutDown()
+		finishedShutdownTest.setFinished()
+	}()
+
+	time.Sleep(20 * time.Millisecond)
```

Comment:
This could be racy in busy CI environments. You'll have to do something more obscure, like adding items until that fails and then verifying that `ShuttingDown()` returns true.